distributed/worker_state_machine.py

```diff
@@ -112,11 +112,14 @@ class SerializedTask(NamedTuple):
     task: object = NO_VALUE
 
 
-class StartStop(TypedDict, total=False):
-    action: str
+class _StartStopRequired(TypedDict):
+    action: Literal["compute", "transfer", "disk-read", "disk-write", "deserialize"]
     start: float
     stop: float
-    source: str  # optional
+
+
+class StartStop(_StartStopRequired, TypedDict, total=False):
+    source: str
```

Comment:
:+1: nice hack

Reply:
I took the hack from https://peps.python.org/pep-0655/#motivation

Reply:
Ah, then please use instead
```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    # TODO import from typing (requires Python >=3.11)
    from typing_extensions import NotRequired

class StartStop(TypedDict):
    action: Literal["compute", "transfer", "disk-read", "disk-write", "deserialize"]
    start: float
    stop: float
    source: NotRequired[str]